### PR TITLE
New Recipe: GoogleCodeSearch v1.2.0

### PR DESCRIPTION
**Pull request by tanmaykm**

### CODE REVIEW

**Comment by tanmaykm** [comment]
Not sure why some builds have failed here. The wizard did not throw any errors when building on my local system before submission.

**Comment by giordano** [comment]
Quoting from https://github.com/JuliaPackaging/Yggdrasil/pull/795#issuecomment-610709831

> Note that we expect the executables to have the appropriate extension (`${exeext}`) for each platforms:
> 
> * `${exeext}` = `".exe"` for Windows
> * `${exeext}` = `""` (empty string) elsewhere
> 
> The `x86_64-w64-mingw32` build is actually successful, but the executable is called `csdp`, without the expected extension.  You should add the end of the build script something like
> ```sh
> if [[ "${target}" == *-mingw* ]]; then
>     mv "${bindir}/csdp" "${bindir}/csdp${exeext}"
> fi
> ```

File: G/GoogleCodeSearch/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/google/codesearch.git", "8ba29bd255b740aee4eb4e4ddb5d7ec0b4d9f23e")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+mkdir -p ${prefix}/share/licenses
+cp codesearch/LICENSE $prefix/share/licenses/
```
Comment (giordano) [inline G/GoogleCodeSearch/build_tarballs.jl:17]: You can just use [`install_license`](https://juliapackaging.github.io/BinaryBuilder.jl/dev/build_tips/#Installing-the-license-file-1) :wink: 
```suggestion
install_license codesearch/LICENSE
```

File: G/GoogleCodeSearch/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+mkdir gsc
+cd gsc
+go mod init github.com/m
+go get -d -v github.com/google/codesearch@v1.2.0
+mkdir -p $prefix/bin
+go build -o $prefix/bin/csearch github.com/google/codesearch/cmd/csearch
+go build -o $prefix/bin/cindex github.com/google/codesearch/cmd/cindex
+exit
```
Comment (giordano) [inline G/GoogleCodeSearch/build_tarballs.jl:25]: In the end the result is the same, but it's better to use the most specific [environment variable](https://juliapackaging.github.io/BinaryBuilder.jl/dev/build_tips/#Automatic-environment-variables-1) when possible: `${bindir}` is the directory where the executables go:
```suggestion
mkdir -p "${bindir}"
go build -o "${bindir}/csearch${exeext}" github.com/google/codesearch/cmd/csearch
go build -o "${bindir}/cindex${exeext}" github.com/google/codesearch/cmd/cindex
```
I've also used `${exeext}` to adjust the extension of the executable on Windows

**Comment by tanmaykm** [comment]
Thanks @giordano! Those were useful to know.
I have added your suggestions.

**Pull request merged**